shoop/campaigns/forms.py

```diff
@@ -77,13 +81,33 @@ def __init__(self, *args, **kwargs):
                     if isinstance(field, OneToOneField):
                         continue
 
+                    self.rule_ids.add(obj.identifier)
                     kwargs = {
-                        "initial": self.get_initial_value(obj.identifier),
+                        "initial": self._get_initial_value(obj),
                         "required": False,  # rules are not required by default, validation is made later on
                         "help_text": obj.description
                     }
-                    self.rule_ids.add(obj.identifier)
-                    self.fields[obj.identifier] = field.formfield(**kwargs)
+                    if isinstance(field, ManyToManyField):
+                        kwargs.update({
+                            "widget": Select2Multiple(model=obj.model)
+                        })
+                        formfield = forms.CharField(**kwargs)
+                        value = kwargs.get("initial")
+                        if value:
+                            formfield.widget.choices = [
+                                (object.pk, object.name) for object in obj.model.objects.filter(pk__in=value)]
+                    else:
+                        formfield = field.formfield(**kwargs)
+                    self.fields[obj.identifier] = formfield
+
+    def _get_initial_value(self, obj):
+        initial_value = self.get_initial_value(obj.identifier)
+        if initial_value:
+            if not isinstance(initial_value, str):
+                initial_value = [o.pk for o in initial_value]
+            else:
+                initial_value = [initial_value]
```

Comment:
Got `TypeError: 'decimal.Decimal' object is not iterable` from the `initial_value = [o.pk for o in initial_value]` line of this.